src/components/Navbar/index.tsx

```diff
@@ -4,37 +4,32 @@ import { connect } from 'react-redux';
 
 import { gettext } from '../../utils';
 import LoginButton from '../LoginButton';
-import { logOutFromServer } from '../../api';
-import { ApplicationState, ConnectedReduxProps } from '../../configureStore';
-import { ApiState } from '../../reducers/api';
 import {
-  User,
-  actions as userActions,
-  getCurrentUser,
-} from '../../reducers/users';
+  ApplicationState,
+  ConnectedReduxProps,
+  ThunkDispatch,
+} from '../../configureStore';
+import { User, getCurrentUser, requestLogOut } from '../../reducers/users';
 import styles from './styles.module.scss';
 
 type PublicProps = {
-  _logOutFromServer: typeof logOutFromServer;
+  _requestLogOut: typeof requestLogOut;
 };
 
 type PropsFromState = {
-  apiState: ApiState;
   profile: User | null;
 };
 
 type Props = PublicProps & PropsFromState & ConnectedReduxProps;
 
 export class NavbarBase extends React.Component<Props> {
   static defaultProps = {
-    _logOutFromServer: logOutFromServer,
+    _requestLogOut: requestLogOut,
   };
 
-  logOut = async () => {
-    const { _logOutFromServer, apiState, dispatch } = this.props;
-
-    await _logOutFromServer(apiState);
-    dispatch(userActions.logOut());
+  logOut = () => {
+    const { _requestLogOut, dispatch } = this.props;
+    (dispatch as ThunkDispatch)(_requestLogOut());
```

Comment:
I saw your comment about this and yes, it would be better if we could find a way to not have to do this type hinting. It seems like the component dispatching the action shouldn't even need to know if it's a "plain" action or a thunk action.